Post for fixing flavor properties. (#993)
SovereignCloudStack/website: _i18n/en/_posts/blog/2024-08-19-flavor-extra-specs-compliance.md

---
layout: post

> title:  "Getting SCS flavor properies into compliance"

Reviewer: ```suggestion
title:  "Getting SCS flavor properties into compliance"
```

category: tech
author:
- "Kurt Garloff"
- "Matthias Büchse"
avatar:
- "garloff.jpg"
about:
- "garloff"
---

## Non-compliance with flavor spec v4

Running the SCS IaaS compatible v4 checks against a cloud may reveal missing
mandatory flavor properties (aka `extra_specs`):
```shell

> garloff@kurt: ~/SCS/standards/Tests$ ./scs-compliance-check.py -a os_cloud=$OS_CLOUD -s REDACTED scs-compatible-iaas.yam

Reviewer: ```suggestion
garloff@kurt: ~/SCS/standards/Tests$ ./scs-compliance-check.py -a os_cloud=$OS_CLOUD -s REDACTED scs-compatible-iaas.yaml
```

Author: Fixed in https://github.com/SovereignCloudStack/website/pull/993/commits/9f8843a57368eebe31236df2f67566ba1b02e7fe, thanks!

WARNING: Flavor 'SCS-1V-4' found via name only, missing property 'scs:name-v2'
ERROR: Flavor 'SCS-1V-4' violating property constraints: cpu-type: None should be 'shared-core'; name-v1: None should be 'SCS-1V:4'; nam e-v2: None should be 'SCS-1V-4'
[...]
WARNING: Flavor 'SCS-1L-1-5' found via name only, missing property 'scs:name-v2'
ERROR: Flavor 'SCS-1L-1-5' violating property constraints: cpu-type: None should be 'crowded-core'; name-v1: None should be 'SCS-1L:1:5'
; name-v2: None should be 'SCS-1L-1-5'
DEBUG: Total critical / error / info: 0 / 28 / 0
********************************************************************************
REDACTED SCS-compatible IaaS v4 (effective):
- main: FAIL (4 passed, 1 failed)
  - FAILED:
    - standard-flavors-check:
      > Must fulfill all requirements of https://docs.scs.community/standards/scs-0103-v1-standard-flavors
```
Let's quickly look at these results: We passed all compliance checks
except for receiving 28 errors on the 15 mandatory and 13 recommended flavors.

## Mandatory flavor properties

The test tool provides the reference to the spec that we need to fulfill:
<https://docs.scs.community/standards/scs-0103-v1-standard-flavors>

> Let's look at the [relevant section](https://docs.scs.community/standards/scs-0103-v1-standard-flavors#properties-extra_specs):

Reviewer: I think it's this section: https://docs.scs.community/standards/scs-0103-v1-standard-flavors/#guarantees-and-properties
```suggestion
Let's look at the [relevant section](https://docs.scs.community/standards/scs-0103-v1-standard-flavors/#guarantees-and-properties):
```
The other one only describes the properties, but this one says they are required for the standard flavors.

Author: Fixed in https://github.com/SovereignCloudStack/website/pull/993/commits/9f8843a57368eebe31236df2f67566ba1b02e7fe

> We lack the `scs:name-v1`, `scs:name-v2`, `scs:cpu-type` and `scs:diskN-type` flavor properties (`extra_specs`).

Reviewer: I think we can put `disk0` here because the standard flavors only require `disk0` (if applicable).
```suggestion
We lack the `scs:name-v1`, `scs:name-v2`, `scs:cpu-type` and `scs:disk0-type` flavor properties (`extra_specs`).
```

Author: Thanks, fixed in https://github.com/SovereignCloudStack/website/pull/993/commits/9f8843a57368eebe31236df2f67566ba1b02e7fe
